### ISSUE TITLE
Full file extentions are not removed from AMD moduleIds

### ISSUE THREAD

**Issue by Harris-Miller** [babel/babel#844]
I have a project where, when using grunt-babel, to keep our Gruntfile.js easy, we decided to name our all es6 files to end with .es6.js, so globbing is easy for compilation. I was sad to find out that when using named AMD, the .es6 was left on in the define('filename.es6'). I was hoping for define('filename'), regardless if the extention is single or multiple.

Would love to see a feature flag added for this under options, something like removeFullModuleIdExtensions:bool

here is a cli example:

./src/myfile.es6.js
```
var module = "I am an AMD module!";

export default module;
```
Then run:
`λ babel myfile.es6.js --out-file ../dist/myfile.js -M -m amd`

Gives you ./dist/myfile.js
```
define("myfile.es6", ["exports", "module"], function (exports, module) {
  "use strict";

  var _module = "I am an AMD module!";

  module.exports = _module;
});
```

would like to to be:
`define("myfile", ["exports", "module"], function (exports, module) {`

I couldn't find a way to do this in the current options available. And a quick dig into the source code didn't show me a way either. if there is already a way to do that, can you point me to that direction. thanks!

**Comment by sebmck** [comment]
The behaviour was previously to remove everything after the first dot but was changed with #674.

**Comment by sebmck** [comment]
I'll probably just end up adding an `getModuleName` option where you can specify a function instead of a `removeFullModuleIdExtensions` option because there are so many different variations that people may want.

**Comment by sebmck** [comment]
Added as of 4.4.0, thanks!

**Comment by sebmck** [comment]
Actually, just removed the `getModuleName` option it's pretty pointless. You can now specify a custom id via `moduleId` ie. `require("babel").transform("code", { modules: "amd", moduleId: "my custom id here" })`. Available as of 4.4.2
